tests/core/pyspec/eth2spec/test/gloas/block_processing/test_process_proposer_slashing.py

```diff
@@ -0,0 +1,110 @@
+import random
+
+from eth2spec.test.context import (
+    spec_state_test,
+    with_gloas_and_later,
+)
+from eth2spec.test.gloas.epoch_processing.test_process_builder_pending_payments import (
+    create_builder_pending_payment,
+)
+from eth2spec.test.helpers.proposer_slashings import (
+    get_valid_proposer_slashing,
+)
+from eth2spec.test.helpers.state import next_epoch
+from eth2spec.test.phase0.block_processing.test_process_proposer_slashing import (
+    run_proposer_slashing_processing,
+)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_builder_payment_deletion_current_epoch(spec, state):
+    """Test that builder pending payment is deleted when proposer is slashed in the same epoch as the proposal."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+    slashed_slot = state.slot + random.randrange(spec.SLOTS_PER_EPOCH)
+
+    # Create a proposer slashing for the current slot
+    proposer_slashing = get_valid_proposer_slashing(
+        spec, state, slot=slashed_slot, signed_1=True, signed_2=True
+    )
+    slashed_proposer_index = proposer_slashing.signed_header_1.message.proposer_index
+
+    # Verify the slashing is for the current epoch
+    assert spec.compute_epoch_at_slot(slashed_slot) == spec.get_current_epoch(state)
+
+    # Add a builder pending payment for the slashed proposer in the current epoch
+    payment_amount = spec.MIN_ACTIVATION_BALANCE
+    payment_weight = 1000
+    fee_recipient = b"\x42" * 20
+
+    payment = create_builder_pending_payment(
+        spec, slashed_proposer_index, payment_amount, payment_weight, fee_recipient
+    )
+    # Place in current epoch slot (SLOTS_PER_EPOCH + slot % SLOTS_PER_EPOCH)
+    slashed_slot_index = spec.SLOTS_PER_EPOCH + slashed_slot % spec.SLOTS_PER_EPOCH
+    state.builder_pending_payments[slashed_slot_index] = payment
+
+    yield from run_proposer_slashing_processing(spec, state, proposer_slashing)
```

Comment:
Kinda weird that we use a slashing for a slot that is in the future from the point of view of the state, but the test is formally correct anyway so it LGTM.